ts/defs.bzl

```diff
@@ -172,6 +173,12 @@ def ts_project(
 
             See [docs/transpiler.md](/docs/transpiler.md) for more details.
 
+        declaration_transpiler: A custom transpiler tool to run that produces the TypeScript declaration outputs instead of `tsc`.
```

Comment:
It looks like `declaration_transpiler` is ignored if `declaration` is False?

If so, should mention that in this docstring and also `fail` on that condition to help users out since as it will be an easy thing to miss